// @urbit/vite-plugin-urbit/dist/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
exports.__esModule = true;
exports.urbitPlugin = void 0;
var htmlPlugin = require("vite-plugin-html-config");
var vite_plugin_rewrite_all_1 = require("vite-plugin-rewrite-all");
var UrbitProxyPlugin = function (_a) {
    var base = _a.base, target = _a.target, options = __rest(_a, ["base", "target"]);
    var basePath = "/apps/" + base + "/";
    return {
        name: "return-partial",
        config: function () {
            var _a;
            return ({
                base: basePath,
                server: {
                    proxy: (_a = {},
                        _a["^" + basePath + "desk.js"] = __assign({ target: target }, options),
                        _a["^((?!" + basePath + ").)*$"] = __assign({ target: target }, options),
                        _a)
                }
            });
        }
    };
};
/**
 * Setup a vite dev server for urbit development
 *
 */
var urbitPlugin = function (config) {
    var htmlPluginOpt = {
        headScripts: [
            { src: "/apps/" + config.base + "/desk.js" },
            { src: "/session.js" },
        ]
    };
    return [UrbitProxyPlugin(config), (0, vite_plugin_rewrite_all_1["default"])(), htmlPlugin(htmlPluginOpt)];
};
exports.urbitPlugin = urbitPlugin;
